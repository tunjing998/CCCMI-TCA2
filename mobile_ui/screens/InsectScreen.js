import React, {useEffect, useState} from 'react';
import {View, StyleSheet, Image} from 'react-native';
import {useTheme} from '@react-navigation/native';
import {Text, Button} from 'react-native-elements';
import LinearGradient from 'react-native-linear-gradient';
import testVariables from '../appium_automation_testing/test_variables';
import axios from 'axios';

const InsectScreen = ({navigation, route}) => {
  const {colors} = useTheme();
  const theme = useTheme();
  const [insectList, setInsectList] = useState([]);
  const [image, setImage] = useState('');

  const styles = StyleSheet.create({
    container: {
      flex: 1,
      alignItems: 'center',
      justifyContent: 'center',
    },
    desc: {
      marginBottom: 20,
      color: colors.text,
    },
    tinyLogo: {
      width: 80,
      height: 80,
    },
  });

  const renderInsect = () => {
    if (insectList.length > 0) {
      console.log(insectList);
      let comp = [];
      comp.push(<Text style={{alignSelf: 'flex-start'}}>Selected Insects:</Text>)
      insectList.map(item => {
        comp.push(
          <View style={{flexDirection: 'row'}}>
            <Image
              style={styles.tinyLogo}
              source={{
                uri: item.insect_image,
              }}
            />
            <Text
              style={{
                fontSize: 15,
                width: 150,
                textAlign: 'center',
                color: colors.text,
              }}>
              {item.insect_name}
            </Text>
            <Text
              style={{
                fontSize: 15,
                width: 150,
                textAlign: 'center',
                color: colors.text,
              }}>
              {item.amount}
            </Text>
          </View>,
        );
      });
      return comp;
    }
  };

  useEffect(() => {
    if (route.params?.post) {
      setInsectList(route.params.post);
    }
  }, [route.params?.post]);

  return (
    <View
      style={styles.container}
      accessibilityLabel={testVariables.insectScreenContainer}
      testID={testVariables.insectScreenContainer}>
      {renderInsect()}
      <Button
        title="Select Insect"
        onPress={() => navigation.navigate('selectInsect1')}
        accessibilityLabel={testVariables.insectScreenSelectInsectButton}
        testID={testVariables.insectScreenSelectInsectButton}
        ViewComponent={LinearGradient}
        linearGradientProps={{
          colors: ['#4c4cff', '#6666ff'],
          start: {x: 0, y: 0},
          end: {x: 0, y: 1.5},
        }}
        buttonStyle={{
          margin: 5,
          padding: 20,
          borderRadius: 20,
          width: 300,
        }}
      />
      <Button
        accessibilityLabel={testVariables.insectScreenAnalyzeInsectButton}
        testID={testVariables.insectScreenAnalyzeInsectButton}
        title="Analyze Insect"
        onPress={() => {navigation.navigate('AnalyzeInsect')}}
        ViewComponent={LinearGradient}
        linearGradientProps={{
          colors: ['#4c4cff', '#6666ff'],
          start: {x: 0, y: 0},
          end: {x: 0, y: 1.5},
        }}
        buttonStyle={{
          margin: 5,
          padding: 20,
          borderRadius: 20,
          width: 300,
        }}
      />
      <Button
        title="Finish"
        buttonStyle={{
          backgroundColor: 'lightgreen',
          marginTop: 30,
          borderRadius: 10,
          width: 100,
        }}
        titleStyle={{color: 'black'}}
      />
    </View>
  );
};

export default InsectScreen;
