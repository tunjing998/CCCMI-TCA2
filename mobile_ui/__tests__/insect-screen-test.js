import React from 'react';
import renderer from 'react-test-renderer';
import InsectScreen from '../screens/InsectScreen';

import {NativeModules as RNNativeModules} from 'react-native';
RNNativeModules.UIManager = RNNativeModules.UIManager || {};
RNNativeModules.UIManager.RCTView = RNNativeModules.UIManager.RCTView || {};
RNNativeModules.RNGestureHandlerModule = RNNativeModules.RNGestureHandlerModule || {
  State: {BEGAN: 'BEGAN', FAILED: 'FAILED', ACTIVE: 'ACTIVE', END: 'END'},
  attachGestureHandler: jest.fn(),
  createGestureHandler: jest.fn(),
  dropGestureHandler: jest.fn(),
  updateGestureHandler: jest.fn(),
};
RNNativeModules.PlatformConstants = RNNativeModules.PlatformConstants || {
  forceTouchAvailable: false,
};

test('insect screen snapShot', () => {
  const snap = renderer.create(<InsectScreen />).toJSON();
  expect(snap).toMatchSnapshot();
});
