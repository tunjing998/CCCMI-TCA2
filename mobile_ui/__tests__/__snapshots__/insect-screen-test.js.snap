// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`insect screen snapShot 1`] = `
<View
  accessibilityLabel="insectScreenContainer"
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="insectScreenContainer"
>
  <Text
    style={
      Object {
        "color": "rgb(28, 28, 30)",
        "marginBottom": 20,
      }
    }
  >
    Your Insects found section is empty!
  </Text>
  <Text
    style={
      Object {
        "color": "rgb(28, 28, 30)",
        "marginBottom": 20,
      }
    }
  >
    If insects is found, choose one of the method below to start.
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 3,
        "overflow": "visible",
        "shadowColor": "rgba(0,0,0, .4)",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      }
    }
  >
    <View
      accessibilityLabel="insectScreenSelectInsectButton"
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="insectScreenSelectInsectButton"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0.5,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
            "paddingHorizontal": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2089dc",
              "fontSize": 20,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Select Insect
        </Text>
      </View>
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#242424",
      }
    }
  >
    
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 3,
        "overflow": "visible",
        "shadowColor": "rgba(0,0,0, .4)",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 1,
        "shadowRadius": 1,
      }
    }
  >
    <View
      accessibilityLabel="insectScreenAnalyzeInsectButton"
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="insectScreenAnalyzeInsectButton"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#2089dc",
            "borderRadius": 3,
            "borderWidth": 0.5,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 8,
            "paddingHorizontal": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2089dc",
              "fontSize": 20,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Analyze Insect
        </Text>
      </View>
    </View>
  </View>
</View>
`;
